src/visGeometry/index.ts

```diff
@@ -1060,21 +1060,43 @@ class VisGeometry {
         }
     }
 
+    /**
+     * Agent color handling
+     * @property this.colorsData is an array of floats, each 4 floats is a color
+     * @param dataColorIndex is an index into the colorsData array, so it is a multiple of 4
+     * @param colorId is an index into the numberOfColors, so it is a number between 0
+     * and numberOfColors-1.
+     * @property this.idColorMapping uses @param colorId. It maps agent id to colorId
+     *
+     * No other module should know about this.colorsData, or the fact that it's 4 times as
+     * long as the input colors. They should only know about colorId
+     * Therefore all the public color methods should use colorId, and the private methods
+     * can convert between colorId and dataColorIndex
+     */
+
+    /**
+     * Agent color handling: private methods
+     */
+
+    private get numberOfColors(): number {
```

Comment:
Nit: I feel this would be more readable when it is used later in the code, if it were a simple function.  It always messes me up when I think it's a simple variable and then wonder, who is updating its value when we add a new color??